### PR TITLE
Allow to configure syslog port / hostname / dest file and json format 

### PR DESCRIPTION
**Pull request by JulienPalard**
That's the version we actually use in production, our syslog daemon listens on /dev/log so we had to modify it. Also we're using json format as it's fowarded to ES so we had to modify it.

There's still an issue: we don't know how to make a configuration optional using pm2 module configuration, if we set "null" pm2 crashes (https://github.com/Unitech/pm2/issues/2972), we can't set undefined as cli-table would expect (undefined does not exist in json) so … I don't know (and don't really care as null looks the right answer, let's wait for pm2 or cli-table to fix it?).

### CODE REVIEW

**Comment by JulienPalard** [comment]
Oh and I had to package it on npmjs to use it as `pm2 install git+https://…` crashes too, so I deployed this version to https://www.npmjs.com/package/pm2-syslog2 as it looked the only solution to have something usable, sorry it looks bad, feels bad, and is probably just bad.

File: README.md [hunk 1]
```diff
@@ -1,34 +1,28 @@
...
+$ pm2 set pm2-syslog:hostname localhost  # localhost is the default.
+$ pm2 set pm2-syslog:port 514  # 514 is the default.
+# or
+$ pm2 set pm2-syslog:path /dev/log
+
+# Optionally change the facility
+$ pm2 set pm2-syslog:facility local0  # user is the default
 
-# change the default hostname using
-$ pm2 set pm2-syslog:hostname localhost
```
Comment (tw3rp) [inline README.md:28]: @JulienPalard  Why did you remove host name. That is one of the identifier I was using to parse the origin

File: README.md [hunk 2]
```diff
@@ -1,34 +1,28 @@
...
-
-```
-$ sudo service rsyslog restart
-```
 
 ## Install module
 
 ```
 # Install
-$ pm2 install pm2-syslog
```
Comment (tw3rp) [inline README.md:25]: - I have been using git+https://  and seems to be working fine

**Comment by tw3rp** [comment]
@JulienPalard A few points:
- I have been forwarding the logs to elastic search logstash stack (ELK). Our developers user different formats so I use the logs in the application logic
- Can you use a know application like log4js https://github.com/nomiddlename/log4js-node

Lemme know if it sounds okay?

File: README.md (hunk 2 above)
Comment (JulienPalard) [reply README.md:25]: Didn't worked for me, see https://github.com/Unitech/pm2/issues/2973 I'll have to go deeper to find out why, but I was already yacc shaving so I skipped it and only opened an issue

File: README.md (hunk 1 above)
Comment (JulienPalard) [reply README.md:28]: Didn't removed it, it's just a few line before. Also localhost is now the default in package.json.

**Comment by JulienPalard** [comment]
We already have the whole stack to do syslog → ES in other applications we already have in production in other languages, so it seemed natural to just push to syslog and have your existing stack do the rest.

**Comment by JulienPalard** [comment]
If you use your pm2-syslog, beware, as null was not allowed in package.json `config`, I had to put a `path`, so it will change the behavior of the current implementation. I'll wait for https://github.com/Unitech/pm2/issues/2972 to be fixed to revert https://github.com/tw3rp/pm2-syslog/pull/1/commits/4ec8c175dc243f228ce12c8ea3edecb8594117e6 restoring the original behavior.
